src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs

```diff
@@ -440,8 +439,10 @@ private bool CanContinueBatch(ReplicationState state, ref long next)
             var totalSize =
                 state.Size + state.Context.Transaction.InnerTransaction.LowLevelTransaction.AdditionalMemoryUsageSize.GetValue(SizeUnit.Bytes);
 
+            int numberOfItemsSent = state.ScannedItems - 1;
```

Comment:
this is wrong, there is no guarantee that we send the scanned item.

Reply:
Done